extension/ackextension/inmemory.go

```diff
@@ -0,0 +1,83 @@
+package ackextension
+
+import (
+	"sync"
+)
+
+type InMemoryAckExtension struct {
+	partitionMap sync.Map
+}
+
+func NewInMemoryAckExtension() AckExtension {
+	return &InMemoryAckExtension{}
+}
+
+type ackStatus struct {
+	id     uint64
+	ackMap sync.Map
+}
+
+func newAckStatus() *ackStatus {
+	id := uint64(0)
+	ret := ackStatus{id: id}
+	ret.ackMap.Store(id, false)
+	return &ret
+}
+
+func (as *ackStatus) nextAck() uint64 {
+	as.id += 1
```

Comment:
that doesn't seem thread safe

Reply:
Ah you're right, I changed this to atomic.uint64.